rust-runtime/aws-smithy-http/src/endpoint.rs

```diff
@@ -5,58 +5,40 @@
 
 //! Code for resolving an endpoint (URI) that a request should be sent to
 
+#![allow(deprecated)]
+
+use crate::endpoint::error::InvalidEndpointError;
+use http::uri::{Authority, Uri};
 use std::borrow::Cow;
-use std::fmt::Debug;
 use std::result::Result as StdResult;
 use std::str::FromStr;
 
-use http::uri::{Authority, Uri};
-
-use aws_smithy_types::config_bag::{Storable, StoreReplace};
 pub use error::ResolveEndpointError;
 
-use crate::endpoint::error::InvalidEndpointError;
-
-pub mod error;
-
 /// An endpoint-resolution-specific Result. Contains either an [`Endpoint`](aws_smithy_types::endpoint::Endpoint) or a [`ResolveEndpointError`].
+#[deprecated(since = "0.60.1", note = "Was never used.")]
 pub type Result = std::result::Result<aws_smithy_types::endpoint::Endpoint, ResolveEndpointError>;
 
 /// A special type that adds support for services that have special URL-prefixing rules.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct EndpointPrefix(String);
-impl EndpointPrefix {
-    /// Create a new endpoint prefix from an `impl Into<String>`. If the prefix argument is invalid,
-    /// a [`InvalidEndpointError`] will be returned.
-    pub fn new(prefix: impl Into<String>) -> StdResult<Self, InvalidEndpointError> {
-        let prefix = prefix.into();
-        match Authority::from_str(&prefix) {
-            Ok(_) => Ok(EndpointPrefix(prefix)),
-            Err(err) => Err(InvalidEndpointError::failed_to_construct_authority(
-                prefix, err,
-            )),
-        }
-    }
-
-    /// Get the `str` representation of this `EndpointPrefix`.
-    pub fn as_str(&self) -> &str {
-        &self.0
-    }
-}
-
-impl Storable for EndpointPrefix {
-    type Storer = StoreReplace<Self>;
-}
+#[deprecated(
+    since = "0.60.1",
+    note = "Use aws_smithy_runtime_api::client::endpoint::EndpointPrefix instead."
+)]
+pub type EndpointPrefix = aws_smithy_runtime_api::client::endpoint::EndpointPrefix;
 
 /// Apply `endpoint` to `uri`
 ///
 /// This method mutates `uri` by setting the `endpoint` on it
+#[deprecated(
+    since = "0.60.1",
+    note = "Use aws_smithy_runtime::client::endpoint::apply_endpoint instead."
+)]
 pub fn apply_endpoint(
     uri: &mut Uri,
     endpoint: &Uri,
     prefix: Option<&EndpointPrefix>,
 ) -> StdResult<(), InvalidEndpointError> {
```

Comment:
I think we can actually leave `ResolveEndpointError` and `InvalidEndpointError` in this crate unchanged and make a new version of it in the new crate that this function returns. Then we avoid any semver risks with these error types. This type can be removed in 0.61.

Reply:
Good point. I restored them in cc1a98aff156f04da9aca7db3d61a8c3c94e2407.